beekeeper-path-scheduler-apiary/src/main/java/com/expediagroup/beekeeper/scheduler/apiary/messaging/MessageEventToPathEventMapper.java

```diff
@@ -17,84 +17,141 @@
+    Boolean isExpired = lifeCycleMap.get(EXPIRED).getBoolean(tableParameters);
+
+    // Generate eventModels for the Lifecycle from the current event and if it is referenced or expired.
+    // TODO: This could probably be genericized down more to not have to pass isUnreferenced / isExpired to the gather call.
+    ArrayList<ApiaryLifeCycleEvent.EventModel> eventPaths = eventType.gatherEventPaths(listenerEvent, isUnreferenced, isExpired);
+
+    // Loop through the event paths & convert them to HouseKeepingPaths
+    List<HousekeepingPath> paths = new ArrayList<>();
+    eventPaths.forEach(event -> paths.add(generatePath(event.lifeCycleEvent, listenerEvent, event.cleanupPath)));
+
+    // If we have HouseKeepingPaths, return new PathEvents else - null
+    PathEvents pathEvents = paths.size() > 0 ? new PathEvents(paths, messageEvent) : null;
+    return Optional.ofNullable(pathEvents);
+  }
+
+  /**
+   * Generates a path to clean up via EntityHousekeepingPath.Builder
+   * @param lifeCycleType The type of lifecycle cleanup event this path is
+   * @param listenerEvent The current event we're generating this path for
+   * @param cleanupLocation The path of the current item to cleanup
+   * @return EntityHouseKeepingPath Path object for the given parameters
+   */
+  private EntityHousekeepingPath generatePath(LifeCycleEventType lifeCycleType, ListenerEvent listenerEvent, String cleanupLocation) {
     EntityHousekeepingPath.Builder builder = new EntityHousekeepingPath.Builder()
-        .pathStatus(PathStatus.SCHEDULED)
-        .creationTimestamp(LocalDateTime.now())
-        .cleanupDelay(extractCleanupDelay(listenerEvent))
-        .clientId(CLIENT_ID)
-        .tableName(listenerEvent.getTableName())
-        .databaseName(listenerEvent.getDbName());
-
-    switch (eventType) {
-    case ALTER_PARTITION:
-      AlterPartitionEvent alterPartitionEvent = (AlterPartitionEvent) listenerEvent;
-      builder.path(alterPartitionEvent.getOldPartitionLocation());
-      break;
-    case ALTER_TABLE:
-      AlterTableEvent alterTableEvent = (AlterTableEvent) listenerEvent;
-      builder.path(alterTableEvent.getOldTableLocation());
-      break;
-    case DROP_PARTITION:
-      DropPartitionEvent dropPartitionEvent = (DropPartitionEvent) listenerEvent;
-      builder.path(dropPartitionEvent.getPartitionLocation());
-      break;
-    case DROP_TABLE:
-      DropTableEvent dropTableEvent = (DropTableEvent) listenerEvent;
-      builder.path(dropTableEvent.getTableLocation());
-      break;
-    default:
-      log.error("Unexpected event type: " + eventType);
-      break;
-    }
+            .pathStatus(PathStatus.SCHEDULED)
+            .creationTimestamp(LocalDateTime.now())
+            .cleanupDelay(extractCleanupDelay(listenerEvent, lifeCycleType))
+            .lifeCycleType(lifeCycleType.name())
+            .apiaryEventType(listenerEvent.getEventType().name())
```

Comment:
Why are we now saving the event type with the path?

Reply:
The idea was for being able to have a distinction between a `CREATE_TABLE` vs `CREATE_PARTITION` expiration for cleanup since it seemed like those cases needed to be handled separately. 

My first thought on this was to make it a boolean flag (i.e. `isTable`), but was concerned if there were any future features we were to add that needed to drive different clean up types based on the apiary event, we could capture it w/ the event and not have to alter it later on.

I'm unsure if there is an existing/preferred way to make that distinction, but I can change this as needed.

Reply:
What is the distinction? As I understand it, `CREATE_TABLE` can create a new expired path entry, and `CREATE_PARTITION` can only update the expiration time on one that already exists.

Reply:
My mistake - I misread the documentation. There shouldn't be a difference there, removing the addition